trio/_dtls.py

```diff
@@ -1207,13 +1252,20 @@ def __exit__(
     ) -> None:
         return self.close()
 
-    def _check_closed(self):
+    def _check_closed(self) -> None:
         if self._closed:
             raise trio.ClosedResourceError
 
+    # async_fn cannot be typed with ParamSpec, since we don't accept
+    # kwargs. Can be typed with TypeVarTuple once it's fully supported
```

Comment:
For what it's worth we *can* enforce that there's a `DTLSChannel` in the first positional argument via ParamSpec. It's probably not worth the trouble, though.

Umm, something like this IIRC?
```py
import typing

P = typing.ParamSpec("P")
R = typing.TypeVar("R")
Stage1 = typing.Callable[typing.Concatenate[DTLSChannel, P], typing.Awaitable[R]]
Stage2 = Stage1[..., R]
```

Then, just `async_fn: Stage2[object]`.

Reply:
Making the signature incomprehensible w/o reading the context does not seem worth it, at least here, but very fancy!

Reply:
Yup. Agreed. FWIW there's an open mypy issue about loosening the syntax to not require type aliasing, but I don't remember there being consensus about it.